### PR TITLE
feat: Add sorting feature [#85]

### LINKED ISSUE #85: [FR] Sort options for find_notes
**Please confirm**
- [x] I am running the latest version of this plugin

**Is your feature request related to a problem? Please describe.**
I want to add a sort option to find_notes.

**Describe the solution you'd like**
```lua
require('telekasten').find_notes( {sort = { 'modified' }} )
```
sort options: title, created, modified

**Additional context**
In zk, sorting is implemented
https://github.com/mickael-menu/zk-nvim

### PR DESCRIPTION
**Pull request by yutkat**
## Proposed change

I added an option (`sort`) to specify sorting when displaying notes.
Right now I only added the sorting feature by filename(`filename`) and modification(`modified`) date.

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update

## Additional information

- This PR fixes or closes issue: fixes #85

## Checklist

- [x] I am running the **latest** version of the plugin.
- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Stylua (a `.stylua.toml` file is provided)
- [x] The code has been checked with luacheck (a `.luacheckrc` file is provided)
- [x] The `README.md` has been updated according to this change.
- [x] The `doc/telekasten.txt` helpfile has been updated according to this change.

### CODE REVIEW

**Comment by lambtho12** [comment]
Thank you for that PR! Seems really nice. I never thought it would be that simple, otherwise I would have done it myself :smile:

File: lua/telekasten.lua [hunk 1]
```diff
@@ -2054,6 +2070,7 @@ local function on_create(opts, title)
             map("n", "<esc>", picker_actions.close(opts))
             return true
         end,
+        sort = M.Cfg.sort,
```
Comment (lambtho12) [inline lua/telekasten.lua:2073]: Is it really necessary here? This picker is for template selection, so it does not really make sense to have it by modified (imo).

File: lua/telekasten.lua [hunk 2]
```diff
@@ -1981,6 +1996,7 @@ local function on_create_with_template(opts, title)
             map("n", "<c-i>", picker_actions.paste_link(opts))
             return true
         end,
+        sort = M.Cfg.sort,
```
Comment (lambtho12) [inline lua/telekasten.lua:1999]: Is it really necessary here? This picker is for template selection, so it does not really make sense to have it by modified (imo).

File: lua/telekasten.lua (hunk 1 above)
Comment (yutkat) [reply lua/telekasten.lua]: I put it on, but you're right, I don't seem to need that, so I'm going to delete it.

File: lua/telekasten.lua (hunk 2 above)
Comment (yutkat) [reply lua/telekasten.lua]: deleted

**Comment by lambtho12** [comment]
Awesome! Thank you very much for contributing.

**Pull request merged**